### ISSUE TITLE
libexpo-modules-core.so file size varies greatly(apple silicon?)

### ISSUE THREAD

**Issue by Mfweb** [expo/expo#20672]
### Summary

eas build:
npx eas build --platform android --profile production
filesize(ARM64-v8a): libexpo-modules-core.so  616,624byte

eas build local:
npx eas build --platform android --local --profile production
filesize(ARM64-v8a):libexpo-modules-core.so  9,114,616byte

### What platform(s) does this occur on?

Android

### Environment

MacBook Air (M2)  
expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.19.0 - ~/.nvs/default/bin/node
      npm: 9.2.0 - ~/.nvs/default/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2021.3 AI-213.7172.25.2113.9123335
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^47.0.8 => 47.0.8 
      react: 18.1.0 => 18.1.0 
      react-dom: 18.1.0 => 18.1.0 
      react-native: 0.70.5 => 0.70.5 
    npmGlobalPackages:
      eas-cli: 3.1.1
    Expo Workflow: managed

### Minimal reproducible example

```
npx create-expo-app test-app
cd test-app
```
eas.json:
```
{
  "cli": {
    "version": ">= 2.8.0",
    "promptToConfigurePushNotifications": false
  },
  "build": {
    "development": {
      "developmentClient": true,
      "distribution": "internal"
    },
    "production": {
      "channel": "production",
      "android": {
        "buildType": "apk",
        "gradleCommand": ":app:assembleRelease",
        "env": {
          "ANDROID_SDK_ROOT": "/Users/***/Library/Android/sdk/"
        }
      }
    }
  },
  "submit": {
    "production": {}
  }
}
```
run:
```npx eas build --platform android --local --profile production```
[libexpo-modules-core.so(local).zip](https://github.com/expo/expo/files/10331055/libexpo-modules-core.so.local.zip)

and
```npx eas build --platform android --profile production```
[libexpo-modules-core.so(server).zip](https://github.com/expo/expo/files/10331052/libexpo-modules-core.so.server.zip)

**Comment by Kudo** [comment]
hi there! you need Android NDK setup correctly to strip the debug symbols from the libraries. otherwise, the native libraries inside APK will be large because of debug symbols.

**Comment by Mfweb** [comment]
@Kudo Do I need to set it in eas.json? I set `"gradleCommand": ":app:assembleRelease"` but doesn't work.

**Comment by Kudo** [comment]
you should install NDK locally. maybe through the `sdkmanager`. e.g. `sdkmanager 'ndk;21.4.7075529' 'ndk;24.0.8215888'`

**Comment by Mfweb** [comment]
> you should install NDK locally. maybe through the `sdkmanager`. e.g. `sdkmanager 'ndk;21.4.7075529' 'ndk;24.0.8215888'`
I found the cause of the problem.
https://github.com/expo/expo/pull/17832#issuecomment-1153902036
Problem solved after setting `"ndk": "24.0.8215888"` and 
```
        "env": {
          "ANDROID_SDK_ROOT": "/Users/***/Library/Android/sdk/",
          "ANDROID_NDK_HOME": "/Users/***/Library/Android/sdk/ndk/"
        }
```
